packages/integration-tests/src/package_manager_sanity_checks.test.ts

```diff
@@ -138,20 +155,29 @@ void describe('getting started happy path', async () => {
   });
 
   void it('throw error on win32 using pnpm', async () => {
-    if (packageManager === 'pnpm' && process.platform === 'win32') {
-      await assert.rejects(
-        execa('pnpm', ['create', amplifyAtTag, '--yes'], {
-          cwd: tempDir,
-        }),
-        (error) => {
-          const errorMessage = error instanceof Error ? error.message : '';
-          assert.match(
-            errorMessage,
-            /Amplify does not support PNPM on Windows./
+    await runWithRetry(
+      async () => {
+        if (packageManager === 'pnpm' && process.platform === 'win32') {
+          await assert.rejects(
+            execa('pnpm', ['create', amplifyAtTag, '--yes'], {
+              cwd: tempDir,
+            }),
+            (error) => {
+              const errorMessage = error instanceof Error ? error.message : '';
+              assert.match(
+                errorMessage,
+                /Amplify does not support PNPM on Windows./
+              );
+              return true;
+            }
           );
-          return true;
         }
-      );
-    }
+      },
+      (error) => {
+        return error.message.includes(
+          'Unexpected token \'<\', "<!DOCTYPE "... is not valid JSON'
+        );
```

Comment:
Does PNPM throw this ? This looks related to app pool, not to PNPM.

I'd remove retry from this particular test for now. I don't recall problems with this particular test (as mentioned here https://github.com/aws-amplify/amplify-backend/pull/2209#discussion_r1835039392) .